cypress/e2e/buy-shirt.cy.ts

```diff
@@ -1,26 +1,49 @@
+import {
+  MenuContentPage,
+  ProductsListPage,
+  ShopingCartPage,
+  LoginPage,
+  AddressStepPage,
+  ShippingStepPage,
+  PaymentStepPage,
+} from "../page/index";
+
+const menuContentPage = new MenuContentPage();
+const productsListPage = new ProductsListPage();
+const shopingCartPage = new ShopingCartPage();
+const loginPage = new LoginPage();
+const addressStepPage = new AddressStepPage();
+const shippingStepPage = new ShippingStepPage();
+const paymentStepPage = new PaymentStepPage();
+
 describe("Buy a t-shirt", () => {
-  it("then the t-shirt should be bought", () => {
-    cy.visit("http://automationpractice.com/");
-    cy.get("#block_top_menu > ul > li:nth-child(3) > a").click();
-    cy.get(
-        "#center_column a.button.ajax_add_to_cart_button.btn.btn-default",
-    ).click();
-    cy.get("[style*='display: block;'] .button-container > a").click();
-    cy.get(".cart_navigation span").click();
-
-    cy.get("#email").type("aperdomobo@gmail.com");
-    cy.get("#passwd").type("WorkshopProtractor");
-
-    cy.get("#SubmitLogin").click();
-    cy.get(".cart_navigation button").click();
-    cy.get("#cgv").check();
-    cy.get("#form button").click();
-    cy.get("#HOOK_PAYMENT a.bankwire").click();
-    cy.get("#cart_navigation button").click();
-
-    cy.get("#center_column > div > p > strong").should(
-        "have.text",
-        "Your order on My Store is complete.",
-    );
+  it("then should be bought a t-shirt", () => {
+    const userData = {
+      emailUser: "aperdomobo@gmail.com",
+      emailPassword: "WorkshopProtractor",
+    };
+
+    const response = {
+      successPurchase: "Your order on My Store is complete.",
+    };
+
+    menuContentPage.visitMenuContentPage();
+    menuContentPage.goToTShirtMenu();
+
+    productsListPage.clickAddCartBtn();
+    productsListPage.clickProceedCheckoutBtn();
+
+    shopingCartPage.clickProceedCheckoutBtn();
+
+    loginPage.login(userData.emailUser, userData.emailPassword);
+
+    addressStepPage.clickProceedCheckoutBtn();
+
+    shippingStepPage.checkTermsCheckbox();
+    shippingStepPage.clickProceedCheckoutBtn();
+
+    paymentStepPage.clickBankWireBtn();
+    paymentStepPage.clickProceedPaymentBtn();
+    paymentStepPage.getConfirmationMessage(response.successPurchase);
```

Comment:
as we are calling this method as `getConfirmationMessage`  the idea is not to perform the validation inside the method. So the method just should return the message and then we perform the validation here:
```js
paymentStepPage.getConfirmationMessage().should(
    "have.text",
    "Your order on My Store is complete.",
);
```